diff --git a/shopping_cart.py b/shopping_cart.py
--- a/shopping_cart.py
+++ b/shopping_cart.py
@@ -42,7 +42,7 @@
                 elif offer.offer_type == SpecialOfferType.TWO_FOR_AMOUNT:
                     x = 2
                     if quantity_as_int >= 2:
-                        total = offer.argument * (quantity_as_int / x) + quantity_as_int % 2 * unit_price
+                        total = offer.argument * (quantity_as_int // x) + quantity_as_int % 2 * unit_price
                         discount_n = unit_price * quantity - total
                         discount = Discount(p, "2 for " + str(offer.argument), -discount_n)
 
